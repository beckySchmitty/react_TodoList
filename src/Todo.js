import React from "react";

const Todo = ({id, task, handleRemove}) => {

    const remove = () => {
        handleRemove(id)
    }

    return (
        <div id={id}>
            {task}
            <button onClick={remove}>X</button>
        </div>
    )
}

export default Todo;