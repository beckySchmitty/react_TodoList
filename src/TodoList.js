import React, {useState} from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import { v4 as uuidv4 } from 'uuid';


const TodoList = () => {
    const [todos, setTodos] = useState([{id: uuidv4(), task: "laundry"}])

    const addTodo = newTodo => {
        setTodos([...todos, newTodo])
    }

    const handleRemove = (id) => {
        setTodos([todos.filter(todo => todo.id !== id)])
    }

    return (
        <div>
            <div><NewTodoForm addTodo={addTodo}/></div>
            {todos.map(todo => <div><Todo id={todo.id} key={todo.id} task={todo.task} handleRemove={handleRemove} /></div>)}
        </div>
    )
}

export default TodoList;