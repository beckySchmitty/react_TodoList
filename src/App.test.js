import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from "./Todo"
import TodoList from "./TodoList"
import App from "./App"

// Smoke tests
it("renders Todo without crashing", function () {
  render(<Todo />);
});

it("renders TodoList without crashing", function () {
  render(<TodoList />);
});

it("renders App without crashing", function () {
  render(<App />);
});


// Snapshots
it("matches snapshot", function () {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot", function () {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});



// Todo functionality
it ("should render with one todo", () => {
    const {getByText} = render(<App />);
    const firstTodo = getByText("laundry")
    expect (firstTodo).toBeInTheDocument();
});

it ("should successfully add todo", () => {
    const {getByText, queryByText, getByLabelText} = render(<App />);

    const doesntShowTodo = queryByText("dishes")
    expect(doesntShowTodo).not.toBeInTheDocument();

    const input = getByLabelText("Task")
    fireEvent.change(input, { target: { value: 'dishes' } })
    const button = getByText("Add");
    fireEvent.click(button);
    const newTodo = getByText("dishes")
    expect(newTodo).toBeInTheDocument();

});

it ("should successfully remove todo", () => {
    const {getByText} = render(<App />);
    const firstTodo = getByText("laundry")
    expect (firstTodo).toBeInTheDocument();
    const removeBtn = getByText("X");
    fireEvent.click(removeBtn);
    expect (firstTodo).not.toBeInTheDocument();
});