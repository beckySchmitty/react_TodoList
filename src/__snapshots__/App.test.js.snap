// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="task"
        >
          Task
        </label>
        <input
          id="task"
          name="task"
          value=""
        />
        <button>
          Add
        </button>
      </form>
    </div>
    <div>
      <div
        id="82b77eea-5b4c-4a4a-aa48-c51e10047b9d"
      >
        laundry
        <button>
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 3`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div>
        <form>
          <label
            for="task"
          >
            Task
          </label>
          <input
            id="task"
            name="task"
            value=""
          />
          <button>
            Add
          </button>
        </form>
      </div>
      <div>
        <div
          id="93049a44-af88-463e-bccf-ab3b683601fa"
        >
          laundry
          <button>
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
